src/core/store.ts

```diff
@@ -365,11 +363,7 @@ export const createStore = (
   ): void | Promise<void> => {
     const writePromise = getAtomState(atom)?.w
     if (writePromise) {
-      writePromise.then(() => {
-        writeAtomState(atom, update)
-        flushPending()
-      })
-      return
+      return writePromise.then(() => writeAtomState(atom, update))
```

Comment:
What I meant, is if you can write a test inside core, for the edge cases - Not the particular line. I'm sorry for not being clear.

Reply:
Sorry, I wasn't clear. You made a good point, we don't have a test for this, even if you don't know what it does.
The reason we didn't have a test is because we didn't know the actual use case.
Now, I feel like this isn't necessary at all. I will remove it completely.